pkg/ottl/ottlfuncs/func_parse_cef_test.go

```diff
@@ -0,0 +1,331 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+
+package ottlfuncs
+
+import (
+	"context"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	"go.opentelemetry.io/collector/pdata/pcommon"
+
+	"github.com/open-telemetry/opentelemetry-collector-contrib/pkg/ottl"
+)
+
+func Test_ParseCEF(t *testing.T) {
+	tests := []struct {
+		name     string
+		input    string
+		expected map[string]any
+		wantErr  bool
+	}{
+		{
+			name:  "basic CEF message",
+			input: "CEF:0|Vendor|Product|1.0|100|Event Name|High|src=10.0.0.1 dst=10.0.0.2",
+			expected: map[string]any{
+				"version":            "0",
+				"deviceVendor":       "Vendor",
+				"deviceProduct":      "Product",
+				"deviceVersion":      "1.0",
+				"deviceEventClassId": "100",
+				"name":               "Event Name",
+				"severity":           "High",
+				"extensions": map[string]any{
+					"src": "10.0.0.1",
+					"dst": "10.0.0.2",
+				},
+			},
+		},
+		{
+			name:  "CEF with escaped pipes in header",
+			input: `CEF:0|Vendor\|Corp|Product\|Name|1.0|100|Event\|Name|High|src=10.0.0.1`,
+			expected: map[string]any{
+				"version":            "0",
+				"deviceVendor":       "Vendor|Corp",
+				"deviceProduct":      "Product|Name",
+				"deviceVersion":      "1.0",
+				"deviceEventClassId": "100",
+				"name":               "Event|Name",
+				"severity":           "High",
+				"extensions": map[string]any{
+					"src": "10.0.0.1",
+				},
+			},
+		},
+		{
+			name:  "CEF with escaped equals in extensions",
```

Comment:
I think the current code is not properly escaping `=` in extensions when there's empty spaces on it, for example: `CEF:0|Vendor|Product|1.0|100|Event|High|key=value is \= to 2`. I think that's a valid use-case, correct?